src/main.rs

```diff
@@ -1,24 +1,88 @@
-use nix::unistd::{Uid, Gid, Group};
-use std::process::exit;
+use nix::libc::{gid_t, uid_t};
+use nix::unistd::{Gid, Uid};
+
 use std::fs;
 use std::io::{self, Write};
+use std::process::exit;
+
+const NEW_USER: &str = "docker_user";
+const NEW_GROUP: &str = "docker_user";
+
+/*
+ * Adds the current uid/gid to /etc/passwd and /etc/shadow
+ */
+fn add_docker_user() -> io::Result<()> {
+    let uid: uid_t = uid_t::from(Uid::current());
+    let gid: gid_t = gid_t::from(Gid::current());
```

Comment:
`Uid`/`Gid` both implement display in such a way that you don't need to convert them to primitive types. to `writeln` them. Though I don't really believe in standard library display functions b/c semantics like "displays the thing" are really vague and might go wrong in the future.

Reply:
So like consider simplify but also consider not simplifying.

Reply:
yeah, I wanted to be sure of the numeric value since it's being put directly inside the /etc files..

Reply:
(I also found it annoying that I couldn't access the numeric values once they went into `Uid` / `Gid` -- raw value is not public).

Reply:
There is `.as_raw()` and also what you did.